app/src/main/java/com/example/tasky/agenda_details/presentation/AgendaDetailsRoot.kt

```diff
@@ -70,10 +80,23 @@ fun AgendaDetailsRoot(
             agendaDetailsViewModel.onDateSelected(selectedDate, dialogState, timeType)
         }
 
+    val compressAndAddImage: (Context, List<Uri>) -> Unit =
+        { context, uri ->
+            agendaDetailsViewModel.compressAndAddImage(context, uri)
```

Comment:
ViewModels should also not receive references to the context, since then you won't be able to write local unit tests anymore. The moment you have an Android SDK reference (like Context) in a class, you need to test it on a real Android device, build an APK, etc. That's a lot slower than being able to run a test just on the JVM.

Therefore, only include Android references in classes that really need them (typically only the UI and certain classes in data)

Reply:
removing context from viewmodel